Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -6296,6 +6296,24 @@ static BOOL shouldUseHighlightsForMarkedText(NSAttributedString *string)
 
 - (void)setAttributedMarkedText:(NSAttributedString *)markedText selectedRange:(NSRange)selectedRange
 {
+    BOOL hasTextCompletion = ^{
+        // UIKit doesn't include the `NSTextCompletionAttributeName`, so the next best way to detect if this method
+        // is being used for a text completion is to check if the attributes match these hard-coded ones.
+        RetainPtr textCompletionAttributes = @{
```

Comment:
Hm... this is a bit of a hack, but I don't really have a cleaner suggestion in the moment :/

Perhaps we could adjust this to return `YES`, iff. the `markedTextAttributes` _contains_ both a `NSForegroundColorAttributeName` of `systemGrayColor` and `NSBackgroundColorAttributeName` of `clearColor`?

That way, if we clean this up in the future by specifying an new key in this dictionary like `NSTextCompletionAttributeName`, doing so won't regress WebKit and we can then safely remove this logic (in favor of checking for `NSTextCompletionAttributeName`, or the future UIKit equivalent)